packages/evm/src/memory.ts

```diff
@@ -70,7 +70,7 @@ export class Memory {
     const returnBuffer = Buffer.allocUnsafe(size)
     // Copy the stored "buffer" from memory into the return Buffer
 
-    const loaded = Buffer.from(this._store.slice(offset, offset + size))
+    const loaded = this._store.slice(offset, offset + size)
```

Comment:
Had to explicitly check that this could not lead to unwanted writes:

- `slice` is a view point, so if you edit this buffer, you edit the "original" buffer too
- `loaded` is not written to in this context
- The `returnBuffer` is `fill`ed, and this actually copies the value (so if you edit it, it does not change the original)

So this is great :)

Reply:
This is half-great. Martin (Holst Swende) explicitly said, that one of the core things we are doing not optimal in our VM is that we copy things around, while we should use the original pointer to memory (the test this half-fixes was actually that there was huge call data created and we copy this over and over on each repeated call).

So we should rather make sure that we also remove this `loaded` copy over time and double check that there *are* just no writes happening. This will eventually/likely give another huge/significant speed bump if we do this right.

Independent from the approval on this PR, since this only eliminates *one* unnecessary copy. So safe anyhow. 🙂 

Will merge in.

Reply:
(updated the comment above for better readability)

Reply:
I think this is fine, `loaded` is not a copy, it is a view, so this should be a super fast operation.

Reply:
Update: just did the test, these are the numbers of the performance tests when we remove this `returnBuffer` copy (this is not out of the box working yet though and needs subsequent fixes, tested on some mainnet blocks).

So these are the new values:

<img width="786" alt="grafik" src="https://user-images.githubusercontent.com/931137/223545905-bf0dc589-3ca0-4994-a594-6bfd8f343773.png">

And these are the old ones:

<img width="783" alt="grafik" src="https://user-images.githubusercontent.com/931137/223546057-244f39f3-6a8a-4063-a8aa-7874a468220e.png">